### ISSUE TITLE
Logic change in how paths are validated

### ISSUE THREAD

**Issue by liampuk** [asciidoctor/asciidoctor-maven-plugin#545]
Thank you for taking your time to talk with us!

**What is this issue about?**

- [x] Bug report
- [ ] Feature request
- [ ] Question

**Description**

I'm having some issues with AsciiDoctor not being able to find `generated-docs` during my Jenkins build. This seems to be as Jenkins sometimes truncates the workspace folder in the path to start with `_`. Previously this was not a problem as asciiDoctor only checks the file name for an `_` prefix, but after #535 every part of the path is checked for for `_` prefixes.

[old code](https://github.com/asciidoctor/asciidoctor-maven-plugin/blob/2a79b16fda4ad30f78ff86799dfd9ecac7cc832f/src/main/java/org/asciidoctor/maven/process/CustomExtensionDirectoryWalker.java#L27) vs [new code](https://github.com/asciidoctor/asciidoctor-maven-plugin/pull/535/files#diff-1ddfad9b3410bbb92a97bf3928c6a3d6704ced7f7ec9ee40c27df87ebc515741R63)

an example path from a failed build:
`/var/lib/jenkins/workspace/_service_name_develop/service-name-service/src/docs/asciidoc`

Was this intentional? If not can this be reverted to the previous logic?

**Environment information**

* asciidoctor-maven-plugin version: 2.2.0

**Comment by abelsromero** [comment]
> an example path from a failed build:
> `/var/lib/jenkins/workspace/_service_name_develop/service-name-service/src/docs/asciidoc`
> 
> Was this intentional? If not can this be reverted to the previous logic?

Resources outside of the project path is something we should not check. In the meantime is perfectly safe to use v2.1.0 of the plugin with latest Asciidoctorj v2.5.1.

**Comment by abelsromero** [comment]
Since it can affect Jenkins users I am going to do a quick 2.2.1 release.

**Comment by mojavelinux** [comment]
Is it accurate to say that the Maven plugin excludes files that begin with underscore by default? If so, perhaps this should be mentioned in the docs. I looked for it, but I couldn't find anything that clearly stated what the default behavior is. (This is related to this issue I came across this week: https://github.com/asciidoctor/asciidoctor.org/issues/800).

**Comment by abelsromero** [comment]
For the maven plugin we have a note in the `sourceDirectory` https://docs.asciidoctor.org/maven-tools/latest/plugin/goals/process-asciidoc/#configuration
We can improve it for sure, for instance check if "." is also ignored and refer as them as "hidden" if that's the recommended teminology. Anything else you see could be improved here?

![image](https://user-images.githubusercontent.com/5781153/126893345-11ba3a5f-29fd-4247-9f4a-fa67447f8a47.png)

**Comment by mojavelinux** [comment]
Aha. I see it now.

I think this is so central to how the plugin (specifically the process-asciidoc goal) works that I think it should be in a non-normative section. That means either putting this information in the preamble, or a section above "Setup" named "How files are processed". There, it can say that this plugin looks for all AsciiDoc files in the specified directory, but ignores files that are considered non-public, such as internal include files and hidden files.

Do you think we should start a separate issue to track it?

**Comment by abelsromero** [comment]
> Do you think we should start a separate issue to track it?

Sure! Just created it

**Comment by mojavelinux** [comment]
#549
